sdk/tests/batch.test.ts

```diff
@@ -0,0 +1,126 @@
+    token1Address = await FungibleToken.deploy(api, admin, 'Coin', 'COIN', 0n)
+  })
+
+  it('batch deposit and withdraw', async function () {
+    this.timeout(20000)
+    await batchTxs(api, admin, [
+      await GRC20.mintTx(admin.addressRaw, 1000000000000n, token0Address),
+      await GRC20.mintTx(admin.addressRaw, 1000000000000n, token1Address),
+      await GRC20.approveTx(invariant.programId(), 1000000000000n, token0Address),
+      await GRC20.approveTx(invariant.programId(), 1000000000000n, token1Address),
+      await invariant.depositSingleTokenTx(token0Address, 1000000000000n),
+      await invariant.depositSingleTokenTx(token1Address, 1000000000000n)
+    ])
+
+    {
+      const balances = await invariant.getUserBalances(admin.addressRaw)
+      assert.deepEqual(
+        [balances.get(token0Address), balances.get(token1Address)],
+        [1000000000000n, 1000000000000n]
+      )
+      assert.strictEqual(await GRC20.balanceOf(admin.addressRaw, token0Address), 0n)
+      assert.strictEqual(await GRC20.balanceOf(admin.addressRaw, token1Address), 0n)
+      assert.strictEqual(
+        await GRC20.balanceOf(invariant.programId(), token0Address),
+        1000000000000n
+      )
+      assert.strictEqual(
+        await GRC20.balanceOf(invariant.programId(), token1Address),
+        1000000000000n
+      )
+    }
+
+    await batchTxs(api, admin, [
+      await invariant.withdrawSingleTokenTx(token0Address, 1000000000000n),
+      await invariant.withdrawSingleTokenTx(token1Address, 1000000000000n)
+    ])
+
+    {
+      const balances = await invariant.getUserBalances(admin.addressRaw)
+      assert.deepEqual(
+        [balances.get(token0Address), balances.get(token1Address)],
+        [undefined, undefined]
+      )
+      assert.strictEqual(await GRC20.balanceOf(admin.addressRaw, token0Address), 1000000000000n)
+      assert.strictEqual(await GRC20.balanceOf(admin.addressRaw, token1Address), 1000000000000n)
+      assert.strictEqual(await GRC20.balanceOf(invariant.programId(), token0Address), 0n)
+      assert.strictEqual(await GRC20.balanceOf(invariant.programId(), token1Address), 0n)
+    }
+  })
+
+  it('batch errors', async function () {
+    this.timeout(20000)
+    try {
+      await batchTxs(api, admin, [
+        await GRC20.burnTx(admin.addressRaw, 1000000000000n, token0Address),
+        await GRC20.burnTx(admin.addressRaw, 1000000000000n, token1Address),
+        await GRC20.approveTx(invariant.programId(), 1000000000000n, token0Address),
+        await GRC20.approveTx(invariant.programId(), 1000000000000n, token1Address),
+        await invariant.depositSingleTokenTx(token0Address, 1000000000000n),
+        await invariant.depositSingleTokenTx(token1Address, 1000000000000n)
+      ])
+    } catch (e) {
+      assert(e instanceof BatchError)
+      assert.deepEqual(
+        e.failedTxs,
+        new Map([
+          [0, 'Panic occurred: Underflow'],
+          [1, 'Panic occurred: Underflow'],
```

Comment:
Additionally, does the batch here have a guarantee of sequential execution, and will the flow be interrupted if one of the transactions fails?